spec.html

```diff
@@ -20665,6 +21279,230 @@ <h1>Static Semantics: PrototypePropertyNameList</h1>
+      <emu-alg>
+        1. Let _names_ be PrivateBoundIdentifiers of |ClassElementList|.
+        1. Append to _names_ the elements of PrivateBoundIdentifiers of |ClassElement|.
+        1. Return _names_.
+      </emu-alg>
+
+      <emu-grammar>
+        MethodDefinition :
+          ClassElementName `(` UniqueFormalParameters `)` `{` FunctionBody `}`
+          `get` ClassElementName `(` `)` `{` FunctionBody `}`
+          `set` ClassElementName `(` PropertySetParameterList `)` `{` FunctionBody `}`
+
+        GeneratorMethod :
+          `*` ClassElementName `(` UniqueFormalParameters `)` `{` GeneratorBody `}`
+
+        AsyncMethod :
+          `async` ClassElementName `(` UniqueFormalParameters `)` `{` AsyncFunctionBody `}`
+
+        AsyncGeneratorMethod :
+          `async` `*` ClassElementName `(` UniqueFormalParameters `)` `{` AsyncGeneratorBody `}`
+      </emu-grammar>
+      <emu-alg>
+        1. Return PrivateBoundIdentifiers of |ClassElementName|.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-class-definitions-static-semantics-containsarguments" type="sdo" aoid="ContainsArguments">
+      <h1>Static Semantics: ContainsArguments</h1>
+      <emu-grammar>
+        IdentifierReference : Identifier
+      </emu-grammar>
+      <emu-alg>
+        1. If the StringValue of |Identifier| is *"arguments"*, return *true*.
+        1. Else, return *false*.
+      </emu-alg>
+
+      <emu-grammar>
+        FunctionDeclaration : `function` BindingIdentifier `(` FormalParameters `)` `{` FunctionBody `}`
+
+        FunctionDeclaration : `function` `(` FormalParameters `)` `{` FunctionBody `}`
+
+        FunctionExpression : `function` BindingIdentifier? `(` FormalParameters `)` `{` FunctionBody `}`
+      </emu-grammar>
+      <emu-alg>
+        1. Return *false*.
+      </emu-alg>
+
+      <emu-grammar>
+        MethodDefinition :
+          ClassElementName `(` UniqueFormalParameters `)` `{` FunctionBody `}`
+          `get` ClassElementName `(` `)` `{` FunctionBody `}`
+          `set` ClassElementName `(` PropertySetParameterList `)` `{` FunctionBody `}`
+      </emu-grammar>
+      <emu-alg>
+        1. Return the result of ContainsArguments of |ClassElementName|.
+      </emu-alg>
+
+      <emu-grammar>
+        GeneratorMethod : `*` ClassElementName `(` UniqueFormalParameters `)` `{` GeneratorBody `}`
+      </emu-grammar>
+      <emu-alg>
+        1. Return the result of ContainsArguments of |ClassElementName|.
+      </emu-alg>
+
+      <emu-grammar>
+        GeneratorDeclaration : `function` `*` BindingIdentifier `(` FormalParameters `)` `{` GeneratorBody `}`
+
+        GeneratorDeclaration : `function` `*` `(` FormalParameters `)` `{` GeneratorBody `}`
+
+        GeneratorExpression : `function` `*` BindingIdentifier? `(` FormalParameters `)` `{` GeneratorBody `}`
+      </emu-grammar>
+      <emu-alg>
+        1. Return *false*.
+      </emu-alg>
+
+      <emu-grammar>ClassBody : ClassElementList</emu-grammar>
```

Comment:
This case should be removed. `arguments` is visible in class bodies (in computed property names) with the value it takes outside:

```js
(function(){
  class A {
    [console.log(arguments[0])]() {}
  }
})(42);
```
prints `42`.